tmt/utils.py

```diff
@@ -7025,21 +7056,65 @@ def is_key_origin(node: fmf.Tree, key: str) -> bool:
 
 
-def resource_files(path: Union[str, Path], package: Union[str, ModuleType] = "tmt") -> Path:
+def resource_files(
+    path: str,
+    package: Union[str, ModuleType] = "tmt"
+        ) -> importlib.abc.Traversable:
     """
     Helper function to get path of package file or directory.
 
     A thin wrapper for :py:func:`importlib.resources.files`:
-    ``files()`` returns ``Traversable`` object, though in our use-case
-    it should always produce a :py:class:`pathlib.PosixPath` object.
-    Converting it to :py:class:`tmt.utils.Path` instance should be
-    safe and stick to the "``Path`` only!" rule in tmt's code base.
-
-    :param path: file or directory path to retrieve, relative to the ``package`` root.
-    :param package: package in which to search for the file/directory.
-    :returns: an absolute path to the requested file or directory.
-    """
-    return Path(importlib.resources.files(package)) / path  # type: ignore[arg-type]
+    ``files()`` returns ``Traversable`` object that can be either a
+    :py:class:`pathlib.PosixPath` or a :py:class:`importlib.reader.MultiplexedPath`.
+
+    If the final path is a file, go ahead and convert it to a regular ``Path``, otherwise
+    keep it as a traversable in order to properly support MultiplexedPaths.
+
+    Additional search paths are introduced from entry-point definitions
+
+    :param path: file or directory path to retrieve, relative to the ``package``
+    or entry-point's root.
+    :param package: primary package in which to search for the file/directory.
+    :returns: a traversable path to the requested file or directory.
+    """
+    def accumulate_path(paths: list[pathlib.Path],
+                        pkg_path: Union[pathlib.Path,
+                                        MultiplexedPath]) -> None:
+        if isinstance(pkg_path, MultiplexedPath):
+            # The root resources.files can be a MultiplexedPath if it is a namespace
+            paths.extend(pkg_path._paths)
+        elif isinstance(pkg_path, pathlib.Path):
+            # Otherwise it should be a normal Path, just add it as-is
+            paths.append(pkg_path)
+        else:
+            # This should not happen
+            raise TypeError(f"Unexpected type improtlib.resources for package: {package}")
+
+    # Accumulate the base path of the package and entry-points
+    base_paths: list[pathlib.Path] = []
+
+    main_path = importlib.resources.files(package)
+    accumulate_path(base_paths, main_path)
+
+    # Additional resource files can be imported from entry-point
+    entry_point_name = 'tmt.resources'
+
+    entry_point_group = entry_points().select(group=entry_point_name)
+
+    for found in entry_point_group:
+        ep_module = found.load()
+        with suppress(Exception):
+            # TODO: Log errors if entry-point was defined that failed to retrieve base path
+            ep_path = importlib.resources.files(ep_module)
+            accumulate_path(base_paths, ep_path)
```

Comment:
WRT logging messages: the best would probably be a `logger` parameter, an optional one. When called from regular code, the call site should provide a specialized logger - it does not now, that's a matter for another patch, I guess. When not provided, which would be the case of import time calls (and regular code too, for now), we would use `tmt.log.Logger.get_bootstrap_logger()` - IIUIC, it should be available once `tmt.utils` is imported successfully, which means `tmt.log` is also ready, and before anything else gets imported.

WRT logging of exceptions: I think it should be safe & desirable to let them bubble up, as long as these exceptions would not happen while importing `tmt.utils`. Once imported, `tmt.__main__` should be able to call `tmt.utils.show_exception()` for anything raised after that point, including exceptions from `resource_files()` running in import time.

Reply:
Thanks for the hint, I've tested and it seems to be working fine:
```console
warn: Failed to load plugin resources: EntryPoint(name='BREAKME', value='tmt_break', group='tmt.resources')
```
Would need the caching though to be less noisy.

I've split the try...catch a bit because the only expected error is at `ep_module = ep.load()`, anything else is rather weird if it occurs.